### ISSUE TITLE
Remove "itemsToRemove"

### ISSUE THREAD

**Issue by blueseph** [blueseph/mise#7]
deletes in the vdom diffing algo are done by putting the elements into an array and removing them later. this _feels_ awful. surely there's a better way to handle, this right?

**Comment by blueseph** [comment]
to go into more detail, the way the diffing algorithm works is it checks an element's children (using the length of whichever is longest), and checks all children by using an index against that length.

``` 
const length = Math.max(updated.children.length, original.children.length)

for (let i= 0; i < length; i++) {
  //...
}
```

the problem is if you remove in place, you're actually _changing_ the number of children updated has. you can remove one no problem, but the moment you remove half plus one, you overflow the remaining children and the remaining ones don't get deleted.

```
console.log(updated.children.length)
// 8
const cachedLength = updated.children.length;

for (let i = 0; i < cachedLength; i++) {
   updated.children[i].remove();
   console.log(`length: ${updated.children.length}`);
   console.log(`i: ${i}`);
}

// length:  8, 7, 6, 5, 4, 3
// i:       0, 1, 2, 3, 4, 5 [OUT OF BOUNDS]
```

if i recalculate, i won't overflow, but i won't get all the elements.

**Comment by blueseph** [comment]
resolved by 642dfc3
